### ISSUE TITLE
GUI : FED : Login - Change password on first login

### ISSUE THREAD

**Issue by derick-montague** [ibm-openbmc/dev#1648]
## Overview
When a user first logs in, they will be required to change their password. Create flow to force password change on first login. 

## UX Flow / Interaction requirements
N/A –See questions below

## cURL commands

To expire the password ssh to the system and run: `passwd --expire root`

## Notes
- To meet CA legislation, we implemented a feature to require the user to update their password on first login.
- Implementation should use nested router view to achieve
- Need to confirm user authentication status during change password
- May need new backend image from Gunnar
- [Angular code reference](https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-webui/+/26436)

## Questions
- [x] Will the backend send something in response that indicates password change required?
  - Nothing sent during initial login request. After successful login, should make a second request to check for PasswordChangeRequired property at /redfish/v1/AccountService/Accounts/${username}
- [x] Are users authenticated after first login–should they be allowed to access the rest of the site?
  - User have limited access to the rest of the site. Most likely navigating to any page will re-route to Unauthorized page 
- [x] User flow unclear. After change password, are users routed to overview screen or do they need to reenter new password at login screen?
  - After successfully changing an expired password, the user should be navigated to the overview page. Do not need to re-enter password
- [x] How should frontend handle authentication state?
  - Authenticated user with expired password has a limited session created
- [x] When user changing password, will min-max password requirements (dynamic values) be available?
  - Min/max password requirements are not accessible.

## References/Resources
This is blocked by the following gerrit review.
-  https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/29136
- [Phosphor-webui story](https://github.com/ibm-openbmc/dev/issues/1049)
- https://github.com/openbmc/docs/blob/master/designs/expired-password.md

## Screenshots

Design for Angular
<img width="1217" alt="Screen Shot 2020-05-18 at 9 25 39 AM" src="https://user-images.githubusercontent.com/8097306/82236883-a4542000-98e9-11ea-9cfc-c57d373c4d3d.png">

**Comment by derick-montague** [comment]
Waiting for upstream changes or need specific build.

**Comment by derick-montague** [comment]
Need to check on the backend review completion before this can move forward. As of 4/9/20, it is expected to move forward in the next few weeks. 

Slack Reference: https://ibm-systems-power.slack.com/archives/C0Q6TQP5Z/p1586450143403400?thread_ts=1586437713.389400&cid=C0Q6TQP5Z

**Comment by derick-montague** [comment]
Per @joseph-reynolds 

> The master branch changes for bmcweb to accept an expired password are working and are in review.  These are needed, for example, so the WebUI can put up a dialog box that says “You password is expired and must be changed; please enter a new password”.  This brings the master branch to parity with the OP940 branch. ==> The changes are in review here:

**Comment by derick-montague** [comment]
Per @joseph-reynolds 
> See updates to the “expired password” PasswordChangeRequired  here: https://lists.ozlabs.org/pipermail/openbmc/2020-April/021399.html
Specifically, you no longer need to log out after you change your password.
The next step is to get this change merged into master or a test driver.

> I also want to merge this into master:  https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/27503
Would the webUI like feedback about why a password change was not accepted? (edited)

**Comment by gtmills** [comment]
https://github.com/ibm-openbmc/dev/issues/1049 (closed) was the old story for this work on phosphor-webui

**Comment by yoshiemuranaka** [comment]
Resolved by https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/33955

**Comment by rfrandse** [comment]
https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/33955 merged
